pages/draft/2019-09/migration-notes.md

```diff
@@ -0,0 +1,176 @@
+| ------------------------ | ------------------------ | ------------- | ---------------- | ------------------------------------------------------------------------------------ |
+| `$schema`                | `$schema`                | Core          | Identifier       | Updated to use 2019-09 meta-schema: `"https://json-schema.org/draft/2019-09/schema"` |
+| *not present*            | `$vocabulary`            | Core          | Identifier       | Declares which vocabularies a schema uses. Enables modular extensibility.            |
+| *not present*            | `$recursiveRef`          | Core          | Identifier       | Replaces `$ref` in recursive contexts.                                               |
+| *not present*            | `$recursiveAnchor`       | Core          | Identifier       | Used in base schemas to define recursive entry points.                               |
+| `definitions`            | `definitions`            | Validation    | Object Container | Still supported but now superseded by `$defs` in 2020-12+.                           |
+| `id`                     | `$id`                    | Core          | Identifier       | Already introduced in Draft 6, still used in Draft 2019-09.                          |
+| `examples`               | `examples`               | Annotation    | Informational    | Remains unchanged.                                                                   |
+| `readOnly` / `writeOnly` | `readOnly` / `writeOnly` | Annotation    | Informational    | Remains unchanged.                                                                   |
+
+
+
+### Migration Tutorial
+
+#### Step 1: Update `$schema` to use the new meta-schema
+
+Start by updating your `$schema` declaration to reference the correct Draft 2019-09 meta-schema:
+
+**Before (Draft 7):**
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#"
+}
+```
+
+**After (Draft 2019-09):**
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema"
+}
+```
+
+
+
+#### Step 2: Add `$vocabulary` if extending or customizing the schema
+
+Draft 2019-09 introduces a **vocabulary system** for declaring which keywords are in use. This is mostly relevant if you write custom meta-schemas or validators.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$vocabulary": {
+    "https://json-schema.org/draft/2019-09/vocab/core": true,
+    "https://json-schema.org/draft/2019-09/vocab/applicator": true,
+    "https://json-schema.org/draft/2019-09/vocab/validation": true,
+    "https://json-schema.org/draft/2019-09/vocab/meta-data": true
+  }
+}
+```
+
+This helps tooling understand what features are supported in your schema. For basic schemas, it can be omitted unless you're writing meta-schemas.
+
+
+
+#### Step 3: Use `$recursiveAnchor` and `$recursiveRef` for recursive structures
```

Comment:
I'm not sure the example here really exemplifies dynamic referencing. It is not just about recursive structures but about generic/templated schemas. That said, it's a bit hard to exemplify that well here. I remember Greg wrote a blog post about this. Maybe worth just linking to that?